src/backend/distributed/planner/shard_pruning.c

```diff
@@ -525,6 +452,53 @@ PruneShards(Oid relationId, Index rangeTableId, List *whereClauseList,
 }
 
 
+/*
+ * BuildPruningInstanceList builds and returns a list of prune instances for
+ * given column by using given from given where clause list and function call
+ * object.
+ */
+List *
+BuildPruningInstanceList(Var *column, List *whereClauseList,
+						 FunctionCall2InfoData *compareValueFunctionCall)
+{
+	PruningTreeNode *tree = CreatePruningNode(AND_EXPR);
+
+	PruningTreeBuildContext *treeBuildContext = palloc0(sizeof(PruningTreeBuildContext));
+	treeBuildContext->current = tree;
+	treeBuildContext->partitionColumn = column;
+
+	/* Build logical tree of prunable restrictions and invalid restrictions */
+	BuildPruningTree((Node *) whereClauseList, treeBuildContext);
+
+	/* Simplify logic tree of prunable restrictions */
+	SimplifyPruningTree(tree, NULL);
+
+	ClauseWalkerContext *clauseWalkerContext =
+		CreateClauseWalkerContext(column, whereClauseList, compareValueFunctionCall);
+
+	/* Figure out what we can prune on */
+	PrunableExpressions(tree, clauseWalkerContext);
+
+	return clauseWalkerContext->pruningInstances;
+}
+
+
+/*
+ * CreateClauseWalkerContext creates a ClauseWalkerContext object initialized
+ * by given arguments.
+ */
+static ClauseWalkerContext *
+CreateClauseWalkerContext(Var *column, List *whereClauseList,
+						  FunctionCall2InfoData *compareValueFunctionCall)
+{
+	ClauseWalkerContext *clauseWalkerContext = palloc0(sizeof(ClauseWalkerContext));
+	clauseWalkerContext->partitionColumn = column;
+	clauseWalkerContext->currentPruningInstance = palloc0(sizeof(PruningInstance));
+	clauseWalkerContext->compareValueFunctionCall = *compareValueFunctionCall;
```

Comment:
maybe there is a better way to pass `FunctionCall2InfoData *` to those functions